Fixing header, moving includes and others

edgarcosta opened the pull request:
Did a couple of things:
1-  moved all includes to power_sums.h, and placed a  `#pragma once` on top so we don't have double definitions. 
2-  fixed the flint includes from `#include <foo.h>` to `#include <flint/foo.h>`
3-  added flint to the libraries
4- removed the unnecessary include_dir, as sage must know where flint is. Before wouldn't find it because it was missing the `flint/` in the includes.
5- added and removed some white space througout. (for a first review consider disabling white space diff)

edgarcosta committed:
moving includes to header, using pragma honce, fixing type return on ps_dynamic_split, fixing trailing whitespace and tabs

edgarcosta committed:
add flint to libraries, remove unecessary include

edgarcosta committed:
white space

kedlaya commented:
This is all fine, but I will have to straighten out some conflicts with my side branch repackage_for_sage at some point. The main one is that there I eliminated power_sums.h entirely because apparently Cython doesn't actually need it; but maybe that is not good if I want the code to also be usable as a standalone C library.

edgarcosta committed:
make ps_dynamic_split match header

edgarcosta commented:
I just fixed another pedantic issue in the last commit.

To me doesn't make much to do something like  `cdef extern from "power_sums.c":`
But yes, one can do that, and then have everything defined in power_sums.c.
In this PR, since I removed both `typedef struct ` from power_sum.c, the cython code will still need the the header.
